src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -203,7 +205,10 @@ class BasePaymentsPage extends React.Component {
      * Hide the add payment modal
      */
     hideAddPaymentMenu() {
-        this.setState({shouldShowAddPaymentMenu: false});
+        this.setState({
+            addPaymentMethodButton: null,
```

Comment:
What is the purpose of setting this to `null`?

Reply:
Understood. Not quite an optimization because calling `setState()` triggers a new call to `render()`. If we want to optimize then we should avoid unnecessary calls to `setState()`.

Reply:
We can remove it.

Reply:
I've removed this statement.